lib/spack/spack/version.py

```diff
@@ -204,11 +262,9 @@ def __getitem__(self, idx):
             return self.version[idx]
 
         elif isinstance(idx, slice):
-            # Currently len(self.separators) == len(self.version) - 1
-            extendend_separators = self.separators + ('',)
             string_arg = []
 
-            pairs = zip(self.version[idx], extendend_separators[idx])
+            pairs = zip(self.version[idx], self.separators[idx])
```

Comment:
@tgamblin This seems to be the only place in Spack that uses `version.separators`. I changed the implementation above to make this simpler. Let me know if you want me to undo this change.